src/builders/actions/deploy.ts

```diff
@@ -80,12 +74,32 @@ export default async function deploy(
     context.logger.info(`📦 Running "${azureHostingConfig.app.target}" on "${context.target.project}"`);
 
     const run = await context.scheduleTarget(target);
-    await run.result;
-
-    files = await getFiles(context, filesPath, projectRoot);
-    if (files.length === 0) {
-      throw new Error('Target did not produce any files, or the path is incorrect.');
+    const targetResult = await run.result;
+    if (!targetResult.success) {
+      throw new Error(`Target failed: ${targetResult.error}`);
     }
+    filesPath = targetResult.outputPath as string;
+
+    if (!filesPath) {
+      if (azureHostingConfig.app.path) {
+        context.logger.warn(`Target was executed but does not provide a result file path.
+        Fetching files from the path configured in azure.json: ${azureHostingConfig.app.path}`);
+        filesPath = path.join(projectRoot, azureHostingConfig.app.path);
+        console.log(filesPath);
+      }
+    }
+  } else if (azureHostingConfig.app.path) {
+    context.logger.info(`Fetching files from the path configured in azure.json: ${azureHostingConfig.app.path}`);
+    filesPath = path.join(projectRoot, azureHostingConfig.app.path);
+  }
+
+  if (!filesPath) {
+    throw new Error('No path is configured for the files to deploy.');
```

Comment:
Can we add a hint on how to fix that? 
Eg. "In the `azure.json`, provide a `path` with a value relative to the root of the project."